modules/nix-darwin/templates/default.nix

```diff
@@ -0,0 +1,121 @@
+      '') (attrNames config.sops.templates)}
+  '';
+in {
+  options.sops = {
+    templates = mkOption {
+      description = "Templates for secret files";
+      type = attrsOf (submodule ({ config, ... }: {
+        options = {
+          name = mkOption {
+            type = singleLineStr;
+            default = config._module.args.name;
+            description = ''
+              Name of the file used in /run/secrets-rendered
+            '';
+          };
+          path = mkOption {
+            description = "Path where the rendered file will be placed";
+            type = singleLineStr;
+            default = "/run/secrets-rendered/${config.name}";
+          };
+          content = mkOption {
+            type = lines;
+            default = "";
+            description = ''
+              Content of the file
+            '';
+          };
+          mode = mkOption {
+            type = singleLineStr;
+            default = "0400";
+            description = ''
+              Permissions mode of the rendered secret file in octal.
+            '';
+          };
+          owner = mkOption {
+            type = singleLineStr;
+            default = "root";
+            description = ''
+              User of the file.
+            '';
+          };
+          group = mkOption {
+            type = singleLineStr;
+            default = "staff";
+            defaultText = lib.literalExpression ''config.users.users.''${cfg.owner}.group'';
+            description = ''
+              Group of the file.
+            '';
+          };
+          file = mkOption {
+            type = types.path;
+            default = pkgs.writeText config.name config.content;
+            defaultText = lib.literalExpression ''pkgs.writeText config.name config.content'';
+            example = "./configuration-template.conf";
+            description = ''
+              File used as the template. When this value is specified, `sops.templates.<name>.content` is ignored.
+            '';
+          };
+        };
+      }));
+      default = { };
+    };
+    placeholder = mkOption {
+      type = attrsOf (mkOptionType {
+        name = "coercibleToString";
+        description = "value that can be coerced to string";
+        check = strings.isConvertibleWithToString;
+        merge = mergeEqualOption;
+      });
+      default = { };
+      visible = false;
+    };
+  };
+
+  config = optionalAttrs (options ? sops.secrets)
+    (mkIf (config.sops.templates != { }) {
+      sops.placeholder = mapAttrs
+        (name: _: mkDefault "<SOPS:${hashString "sha256" name}:PLACEHOLDER>")
+        config.sops.secrets;
+
+      launchd.daemons.sops-nix-templates = {
+        script = renderScript;
+        serviceConfig = {
+          KeepAlive = false;
+          RunAtLoad = true;
+        };
+      };
+      system.activationScripts = {
+        sops-nix-templates.text = ''
+          launchctl unload /Library/LaunchDaemons/org.nixos.sops-nix-templates.plist
```

Comment:
Do you have maybe any idea how we show errors in the action phase from sops nix?

Reply:
I do not, any modules i see logs for when activating are because they use activation scripts vs launchd itself.

Home manager for instance I see this being logged: https://github.com/nix-community/home-manager/blob/8f6ca7855d409aeebe2a582c6fd6b6a8d0bf5661/nix-darwin/default.nix#L17

launchctl is bad from what I can see, and the mac version is even more stripped down. Launchd may not be the answer and instead an activation script is instead the way to go with it.

I'll change the install secrets from launchd to script based and see how it goes as the lack of insight into launchd is annoying